crates/datasources/src/object_store/mod.rs

```diff
@@ -198,10 +201,14 @@ pub trait ObjStoreAccess: Debug + Display + Send + Sync {
         let store = self.create_store()?;
         let mut objects = Vec::new();
         for loc in locations {
-            let list = self
-                .list_globbed(&store, &loc.path())
-                .await
-                .map_err(|e| DataFusionError::External(Box::new(e)))?;
+            let mut list = Vec::new();
+            for path in get_resolved_patterns(loc.path().into_owned()) {
+                let sub_list = self
+                    .list_globbed(&store, path)
+                    .await
+                    .map_err(|e| DataFusionError::External(Box::new(e)))?;
```

Comment:
It looks like `ObjectStoreSourceError` should have a `From<>` implemented for DataFusionError, so I would have thought that we wouldn't have needed the map here anymore?

Reply:
I usually avoid implementing from for different enum errors since a DF error might be `External` or `Execution` etc. Mapping feels fine and explicit to me.